tests/test_slow_retrieval_attack.py

```diff
@@ -201,29 +153,53 @@ def setUp(self):
     # We'll skip using updater.refresh since we don't have a server running,
     # and we'll update the metadata locally, manually.
     shutil.rmtree(os.path.join(
-        self.client_directory, self.repository_name, 'metadata', 'current'))
-    shutil.copytree(os.path.join(self.repository_directory, 'metadata'),
-        os.path.join(self.client_directory, self.repository_name, 'metadata',
+        obj.client_directory, obj.repository_name, 'metadata', 'current'))
+    shutil.copytree(os.path.join(obj.repository_directory, 'metadata'),
+        os.path.join(obj.client_directory, obj.repository_name, 'metadata',
         'current'))
 
+
     # Set the url prefix required by the 'tuf/client/updater.py' updater.
     # 'path/to/tmp/repository' -> 'localhost:8001/tmp/repository'.
-    repository_basepath = self.repository_directory[len(os.getcwd()):]
-    url_prefix = \
-      'http://localhost:' + str(self.SERVER_PORT) + repository_basepath
+    repository_basepath = obj.repository_directory[len(os.getcwd()):]
+
+    obj.server_process_handler = utils.TestServerProcess(log=logger,
+        server='slow_retrieval_server.py', timeout=0, extra_cmd_args=[mode])
+
+    logger.info('Slow Retrieval Server process started.')
+
+    # NOTE: Following error is raised if a delay is not long enough:
+    # <urlopen error [Errno 111] Connection refused>
+    # or, on Windows:
+    # Failed to establish a new connection: [Errno 111] Connection refused'
+    # 1s led to occasional failures in automated builds on AppVeyor, so
+    # increasing this to 3s, sadly.
+    time.sleep(3)
```

Comment:
Can we not pass this as a timeout value to `TestServerProcess`, rather than having a `sleep()` here? I thought `wait_for_server` was designed for cases such as this?

cc @jku

Reply:
I am not sure why, but if we use `wait_for_server` with whatever timeout I come up with then the test always fails with:
 
![image](https://user-images.githubusercontent.com/16246778/95991178-231b9d80-0e35-11eb-9e1c-134385e51701.png)

Reply:
I did not originally remove this sleep since one of the tests actually tests for too slow initial response -- thinking that wait_for_server() would break the test...  but now that I actually read the server code again, I think it may be fine for us to remove the sleep, and give a timeout to TestServerProcess.

If no-one else is sure how it works either, then I guess we'll have to test it to see...

Reply:
If I remove the sleep I get the result I have screenshotted above.
I am not sure why, but I don't think we can remove the sleep here...

Reply:
This one is fixed in my last update.